zcash_client_sqlite/src/wallet.rs

```diff
@@ -4024,6 +4024,41 @@ pub(crate) fn prune_nullifier_map(
     Ok(())
 }
 
+pub(crate) fn get_block_range(
+    conn: &rusqlite::Connection,
+    protocol: ShieldedProtocol,
+    commitment_tree_address: incrementalmerkletree::Address,
+) -> Result<Option<Range<BlockHeight>>, SqliteClientError> {
+    let prefix = match protocol {
+        ShieldedProtocol::Sapling => "sapling",
+        ShieldedProtocol::Orchard => "orchard",
+    };
+    let mut stmt = conn.prepare_cached(&format!(
+        "SELECT MIN(height), MAX(height)
+         FROM blocks
+         WHERE {prefix}_commitment_tree_size BETWEEN :min_tree_size AND :max_tree_size"
```

Comment:
This query fails under use: https://github.com/zcash/zcash-devtool/pulls#issuecomment-2716060300

Add a test to ensure that `get_block_range` behaves correctly for the various possible database states it might be applied to.